tests/specs/controllers/MainCtrlSpec.js

```diff
@@ -0,0 +1,94 @@
+    var FILE_DATA = {"/home/Desktop/test.py" : ["import this\n"]};
+    var COALA_JSON = {
+          "logs": [],
+          "results": {
+            "autopep8": [
+              {
+                "additional_info": "",
+                "affected_code": [
+                  {
+                    "end": {
+                      "column": null,
+                      "file": "/home/Desktop/test.py",
+                      "line": 40
+                    },
+                    "file": "/home/Desktop/test.py",
+                    "start": {
+                      "column": null,
+                      "file": "/home/Desktop/test.py",
+                      "line": 33
+                    }
+                  }
+                ],
+                "debug_msg": "",
+                "diffs": {
+                  "/home/Desktop/test.py": "DIFF\n"
+                },
+                "id": 110780194316329970598265416648045230634,
+                "message": "The code does not comply to PEP8.",
+                "origin": "PEP8Bear",
+                "severity": 1
+              }
+            ]
+          }
+        };
+
+    describe('parseCoalaProject()', function() {
+
+        beforeEach(module('coalaHtmlApp'));
+        beforeEach(inject(function ($controller, $injector,  _$rootScope_) {
+            $httpBackend = $injector.get('$httpBackend');
+            rootScope = _$rootScope_;
+            scope = _$rootScope_.$new();
+            $httpBackend.whenGET('data/Constants.json')
+                .respond(200, CONSTANTS);
+            $httpBackend.whenGET(CONSTANTS.data + CONSTANTS.coala)
+                .respond(200, COALA_JSON);
+            $httpBackend.whenGET(CONSTANTS.data + CONSTANTS.file_data)
+                .respond(200, FILE_DATA);
+            $httpBackend.whenGET(CONSTANTS.data + CONSTANTS.files)
+                .respond(200, FILES);
+            $httpBackend.whenGET(CONSTANTS.roothome)
+                .respond(200, ROOTHOME);
+            $httpBackend.flush();
+            $controller('MainCtrl', {
+                $scope: scope,
+                $rootScope: rootScope
+            });
+        }));
+        afterEach(function() {
+          $httpBackend.verifyNoOutstandingExpectation();
+          $httpBackend.verifyNoOutstandingRequest();
+        });
+
+        it('test result count', function() {
+            $httpBackend.flush();
+            expect(rootScope.resultsCount).to.equal(1);
+        });
+        it('test parseResult', function() {
+            $httpBackend.flush();
+            expect(rootScope.resultFiles).to.have.property("/home/Desktop/test.py")
+              .with.length(1);
+            expect(rootScope.resultFiles["/home/Desktop/test.py"][0].start).to.equal(33);
+        });
+        it('test scope.data', function() {
+            $httpBackend.flush();
+            expect(scope.data).to.have.length(1);
+            expect(scope.data[0]).to.equal(rootScope.resultFiles);
```

Comment:
Um here data has only one object i.e resultFiles. So I guess maybe it's okay here?